src/Kubernetes.Controller/Caching/IngressCache.cs

```diff
@@ -32,10 +32,14 @@ public class IngressCache : ICache
 
     public IngressCache(IOptions<YarpOptions> options, IServerCertificateSelector certificateSelector, ICertificateHelper certificateHelper, ILogger<IngressCache> logger)
     {
-        _options = options?.Value ?? throw new ArgumentNullException(nameof(options));
-        _certificateSelector = certificateSelector ?? throw new ArgumentNullException(nameof(certificateSelector));
-        _certificateHelper = certificateHelper ?? throw new ArgumentNullException(nameof(certificateHelper));
-        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        ArgumentNullException.ThrowIfNull(options?.Value);
```

Comment:
Group checks before the assignment

```suggestion
        ArgumentNullException.ThrowIfNull(options?.Value);   ArgumentNullException.ThrowIfNull(certificateSelector);   ArgumentNullException.ThrowIfNull(certificateHelper);
        ArgumentNullException.ThrowIfNull(logger);

        _options = options.Value;
        _certificateSelector = certificateSelector;
        _certificateHelper = certificateHelper;
```